### ISSUE TITLE
On ChromeOS, in full-screen mode, with "shelf" auto-hiding enabled, Termux has a big chunk of unused space at the bottom of its display

### ISSUE THREAD

**Issue by snogglethorpe** [termux/termux-app#1326]
**Problem description**
A clear and concise description of what the problem with the termux app is. You may post screenshots in addition to description.

I'm using Termux on ChromeOS (with Android support), on a Samsung Chromebook Pro, on the Beta channel (ChromeOS version 78.0.3904.70).

When Termux is in "full screen mode" -- no title bar, uses entire screen, as enabled by the dedicated "full screen" key on the keyboard -- there's a big chunk of wasted space at the bottom of the Termux window, about 2cm high, which is entirely black.

This isn't the "extra keys" display; I've disabled that via the side-swipe menu.  If I enable that, the row of extra key buttons is shown _above_ the blank chuck at the bottom of the window.

Increasing or decreasing the font size via two-finger pinching does not change this; the font size correctly changes, and the terminal redisplays, but the unused black chunk remains at the bottom.

I've attached a screenshot; this is pretty basic, just notice that at the bottom there's a big black area (this isn't the Emacs echo/message area, which is only a single line high).

If I _don't_ use the "full screen" mode (so that a title bar is shown at the top of the Termux window), even if I drag the window boundaries so that the window occupies the entire display, the wasted black space goes away, and Termux correctly uses all space down to the bottom of its window.  However, in this case there's of course space wasted due to the titlebar at the top of the Termux window.... ><

I notice that the wasted space is roughly the same as the size of the ChromeOS shelf, so my guess is that Termux is somehow trying to avoid conflicting with the shelf ... however as I have shelf auto-hiding turned on (not sure whether this is the default), the shelf is not displayed most of the time, so Termux really shouldn't be worrying about it.

Thanks!

![Screenshot 2019-10-27 at 12 23 09](https://user-images.githubusercontent.com/398170/67640035-a787b000-f8b4-11e9-932d-73a95112487e.png)

**Steps to reproduce**
Please post all steps that are needed to reproduce the issue.

**Expected behavior**

**Additional information**
Post output of command `termux-info`.
If you are rooted or have access to adb then capture a logcat with `logcat -d "*:W"`, from a adb or root shell.

Updatable packages:
apt/stable 1.4.9-19 i686 [upgradable from: 1.4.9-15]
bash/stable 5.0.11 i686 [upgradable from: 5.0.9]
binutils/stable 2.33.1 i686 [upgradable from: 2.32-5]
ca-certificates/stable 20191016 all [upgradable from: 20190515]
clang/stable 9.0.0 i686 [upgradable from: 8.0.1]
command-not-found/stable 1.42 i686 [upgradable from: 1.39]
coreutils/stable 8.31-8 i686 [upgradable from: 8.31-7]
curl/stable 7.66.0-1 i686 [upgradable from: 7.65.3-6]
dos2unix/stable 7.4.1 i686 [upgradable from: 7.4.0-1]
emacs/stable 26.3-1 i686 [upgradable from: 26.3]
findutils/stable 4.7.0 i686 [upgradable from: 4.6.0-4]
gdb/stable 8.3.1 i686 [upgradable from: 8.3-4]
glib/stable 2.60.7 i686 [upgradable from: 2.60.6-1]
gnutls/stable 3.6.10 i686 [upgradable from: 3.6.9-1]
libandroid-support/stable 25-2 i686 [upgradable from: 25-1]
libcurl/stable 7.66.0-1 i686 [upgradable from: 7.65.3-6]
libexpat/stable 2.2.9 i686 [upgradable from: 2.2.7-1]
libffi/stable 3.2.1-5 i686 [upgradable from: 3.2.1-4]
libgcrypt/stable 1.8.5 i686 [upgradable from: 1.8.4-1]
libgnutls/stable 3.6.10 i686 [upgradable from: 3.6.9-1]
libllvm/stable 9.0.0 i686 [upgradable from: 8.0.1]
m4/stable 1.4.18-3 i686 [upgradable from: 1.4.18-1]
ncurses/stable 6.1.20190928 i686 [upgradable from: 6.1.20190511-7]
ndk-sysroot/stable 20-6 i686 [upgradable from: 20-1]
openssh/stable 8.1p1 i686 [upgradable from: 8.0p1-4]
openssl/stable 1.1.1d-1 i686 [upgradable from: 1.1.1c-2]
strace/stable 5.3 i686 [upgradable from: 5.2]
termux-tools/stable 0.72 all [upgradable from: 0.69-2]
texinfo/stable 6.7 i686 [upgradable from: 6.6-1]
Subscribed repositories:
https://dl.bintray.com/grimler/science-packages-24 science/stable
https://dl.bintray.com/grimler/game-packages-24 games/stable
https://dl.bintray.com/termux/termux-packages-24 stable/main
System information:
Linux localhost 3.18.0-19747-gd0dcdfafc1b5 #1 SMP PREEMPT Mon Oct 21 01:01:50 PDT 2019 i686 Android
Termux-packages arch:
i686
Android version:
9
Device manufacturer:
Google
Device model:
Samsung Chromebook Pro

**Comment by Grimler91** [comment]
I get the same on my tablet for some zoom levels, when I am in portrait mode. Does the problem disappear if you zoom in? (and become worse if you zoom out?)

**Comment by snogglethorpe** [comment]
> I get the same on my tablet for some zoom levels, when I am in portrait mode. Does the problem disappear if you zoom in? (and become worse if you zoom out?)

What do you mean by zoom in/out?

If I use a pinch gesture to make the terminal font size bigger or smaller, the font size does change correctly, but the unused black space at the bottom of the window seems to remain the same regardless of how much I do so...

**Comment by glykos** [comment]
https://support.google.com/chromebook/thread/1777903?hl=en suggests that this may not be a termux-specific issue.

**Comment by Grimler91** [comment]
@snogglethorpe yeah the pinch gesture. I guess it's not the same issue then, thanks for testing. @glykos linked issue seems more related

**Comment by snogglethorpe** [comment]
Note that it may not be _specific_ to Termux, but it does still seem to be a bug with Termux, as other Android apps on ChromeOS do not have this problem (so it's not an inherent flaw in ChromeOS Android support).  Presumably some other apps have this same bug.

**Comment by snogglethorpe** [comment]
BTW, I found that if I (1) disable shelf auto-hiding, and then (2) re-enable it, Termux now uses the entire display.

Not sure what's going on.

One possible fact that may be connected is that even when shelf auto-hiding is _disabled_ generally, it will actually override this and do shelf auto-hiding _while an app is in full-screen mode._

**Comment by snogglethorpe** [comment]
Note that recently this problem seems to have disappeared, but I don't know why...

**Comment by WHYBBE** [comment]
Why does this problem still exist, where is the problem, ChromeOS or Termux or what?😢😢😢
